### ISSUE TITLE
Allow for the Explore rooms feature to automatically browse/add a Matrix server when the server part of the handle is provided

### ISSUE THREAD

**Issue by Porkepix** [vector-im/riot-web#13670]
### Description

When trying to find chans/rooms through the field present in the "Explore room" panel, one could expect the search to be able to browse the directory listing and/or add, whether is it temporary or definitely the said server when the user input the server part of the handle like `:mymatrixinstance.com`.

I found out about this when, long time user of Mozilla's IRC, and now much less present, I wanted to list and search through all of the chans/rooms of their instance just by filling the field with `:mozilla.org`. It wasn't the case and only found out less than a dozen results.

### Steps to reproduce

Try to search for chans/rooms from the "Explore rooms" panel/field for a server/instance different from our own.
It will likely find no results or almost none.

![Screenshot from 2020-05-14 15-17-49](https://user-images.githubusercontent.com/709500/81939083-298fba00-95f6-11ea-8822-d52b443aef07.png)

### Version information

- **Platform**: web (in-browser) or desktop? Both.
- Latest version for both Web and desktop (1.6).

**Comment by aaronraimist** [comment]
Just in case you haven't found the way it actually works, click on the Matrix rooms (matrix.org) button and then Add server and type in mozilla.org

**Comment by Porkepix** [comment]
Yup that's what I was told, but the wording and what the user was supposed to input was pretty unclear to me, which is why I originally opened #13661. Through the discussion, I came to this current issue because I think things could be quite improved.
